array.go

```diff
@@ -75,6 +75,15 @@ func (a *Array) Path(path string) *Value {
 	return jsonPath(a.chain, a.value, path)
 }
 
+// Alias is similar to Value.Alias.
+func (a *Array) Alias(name string) *Array {
+	a.chain.enter("Alias(%s)", name)
+	defer a.chain.leave()
+
+	a.chain.setAlias(name)
+	return newArray(a.chain, a.value)
+}
```

Comment:
Here and in other structs: instead of returning new value, we can just return receiver (`a` in this case).

In httpexpect, methods usually return new value if they return something unrelated (e.g. element of array), otherwise they just return receiver.

This will affect tests:

* new value will be the same as original (we can use assert.Same)
* Alias() wont be present in path

Reply:
Thanks for review.
For example, for `Array`, should I modify following?
```go
// Alias is similar to Value.Alias.
func (a *Array) Alias(name string) *Array {
	a.chain.setAlias(name)
	return a
}
```

Reply:
Yes, but

```
a.chain.enter("Alias(%s)", name)
defer a.chain.leave()
```

is still needed because it will log successful operation (if a logger is used).

Reply:
@gavv
Thanks.
As shown below, if `chain.enter()` is called before `setAlias()`, the set Alias will be deleted by `chain.leave()`.
```go
func (a *Array) Alias(name string) *Array {
	a.chain.enter("Alias(%s)", name)
	defer a.chain.leave()

	a.chain.setAlias(name)
	return a
}
```
```go
	value1 := NewArray(reporter, []interface{}{1, 2}).Alias("foo")
	assert.Equal(t, []string{"Array()"}, value1.chain.context.Path)
	assert.Equal(t, []string{"foo"}, value1.chain.context.AliasedPath) // failed, AliasedPath equal []string{}

```

This may be avoided by changing the order of calls as shown below. 
```go
func (a *Array) Alias(name string) *Array {
	a.chain.setAlias(name)

	a.chain.enter("Alias(%s)", name)
	defer a.chain.leave()
	return a
}
```

```go
	value1 := NewArray(reporter, []interface{}{1, 2}).Alias("foo")
	assert.Equal(t, []string{"Array()"}, value1.chain.context.Path)
	assert.Equal(t, []string{"foo"}, value1.chain.context.AliasedPath) // success
```
Is this policy correct?

Reply:
Oh, I see. Yes, this is a problem. However, it seems that it should be solved with the new chain interface that I just pushed and described here:

* https://github.com/gavv/httpexpect/pull/227#issuecomment-1401846252
* https://github.com/gavv/httpexpect/pull/227#issuecomment-1401852700

With new interface, it'll look like:

```go
func (a *Array) Alias(name string) *Array {
	opChain := a.chain.enter("Alias(%s)", name)
	defer opChain.leave()

	a.chain.setAlias(name)
	return a
}
```

Note that leave is called on opChain, but setAlias is still called on a.chain, since we want to modify array's chain, not operation's chain.

Reply:
Thank you for commits. I fixed on 534ab72 using opChain.